WordPress/openverse-frontend
# Add breakpoint based modal variations

## Fixes


Fixes #522 by @panchovm 
Fixes #1375 by @AetherUnbound 

## Description


Updates the existing modal implementation to add breakpoint-based variations. It's just style updates except that the mobile version also includes the logo button. I added that because the only place we actually currently use the modal is for the mobile filters and search type switcher and those use the logo button in the header.

This does also wrap the top bar in a slot to make it easier to modify the modal content without having to copy/paste the whole thing like we've done with the `VMobileModalContent`.

This PR also removes that component :tada: 

## Testing Instructions

Checkout the branch and run `pnpm dev` to visit the site locally.

Go to the search results page with a small viewport width and verify that the modal for the search type switcher and the filters still works as expected and that it looks good.

Next run `pnpm storybook` and open the Modal story in storybook and test out the various breakpoints using Storybook's breakpoints option or by opening the iframe popout and changing your viewport width manually (the second option can give you a sense of how responsive it is).

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
